src/hotspot/share/classfile/protectionDomainCache.hpp

```diff
@@ -104,20 +102,17 @@ class ProtectionDomainCacheTable : public Hashtable<WeakHandle, mtClass> {
 };
 
 
+// This describes the linked list protection domain for each DictionaryEntry in pd_set.
 class ProtectionDomainEntry :public CHeapObj<mtClass> {
-  friend class VMStructs;
- public:
-  ProtectionDomainEntry* _next;
   ProtectionDomainCacheEntry* _pd_cache;
+  ProtectionDomainEntry* volatile _next;
+ public:
 
-  ProtectionDomainEntry(ProtectionDomainCacheEntry* pd_cache, ProtectionDomainEntry* next) {
-    _pd_cache = pd_cache;
-    _next     = next;
-  }
+  ProtectionDomainEntry(ProtectionDomainCacheEntry* pd_cache,
+                        ProtectionDomainEntry* head) : _pd_cache(pd_cache), _next(head) {}
 
-  ProtectionDomainEntry* next() { return _next; }
-  void set_next(ProtectionDomainEntry* entry) { _next = entry; }
-  oop object();
+  ProtectionDomainEntry* next() { return Atomic::load(&_next); }
+  void set_next(ProtectionDomainEntry* entry) { Atomic::store(&_next, entry); }
```

Comment:
Again these may need to be load_acquire/release_store for correct lock-free traversal. With suitable name changes for the methods.

Reply:
Thanks, I'll add acquire/release and change the names to match the implementation details.

Reply:
Did we collectively agree on the naming convention that set_next has to be release_set_next because the implementation uses release_store, or is this just your preference?  I find this noisy in the context where these functions are being called.